### PR TITLE
GLTFExporter: add onError parameter to parse

### PR DESCRIPTION
**Pull request by marcofugaro**
Related issue: https://github.com/mrdoob/three.js/pull/22755#issuecomment-955798932

**Description**

This allows us to correctly catch errors, even in the promise after `this.pending`.

I did this by converting `GLTFWriter.write()` to async since it was cleaner.

### CODE REVIEW

**Comment by Mugen87** [comment]
This change introduces a syntax error which breaks the exporter:

> Uncaught SyntaxError: Identifier 'options' has already been declared

https://raw.githack.com/marcofugaro/three.js/gltfexporter-errors/examples/misc_exporter_gltf.html

**Comment by marcofugaro** [comment]
Ups, thanks for checking!

Fixed now.

**Comment by Mugen87** [comment]
It seems the documentation page also needs a small update: https://threejs.org/docs/index.html#examples/en/exporters/GLTFExporter

File: examples/jsm/exporters/GLTFExporter.js [hunk 1]
```diff
@@ -94,9 +94,17 @@ class GLTFExporter {
...
+	 * @param  {Function} onError  Callback on errors
 	 * @param  {Object} options options
 	 */
-	parse( input, onDone, options ) {
+	parse( input, onDone, onError, options ) {
+
+		// legacy: the third argument used to be options
+		if ( typeof onError === 'object' ) {
+
+			options = onError;
```
Comment (Mugen87) [inline examples/jsm/exporters/GLTFExporter.js:105]: Probably better to add a warning here? Something like:
```js
console.warn( 'THREE.GLTFExporter: parse() expects options as the fourth argument now.' );
```
This change also needs to be noted in the migration guide.

**Comment by Mugen87** [comment]
> Is it okay like this?

That looks good to me!

File: examples/jsm/exporters/GLTFExporter.js (hunk 1 above)
Comment (marcofugaro) [reply examples/jsm/exporters/GLTFExporter.js:106]: Done.

> This change also needs to be noted in the migration guide.

I have no access to the wiki to make this change, can you take care of it?

**Comment by marcofugaro** [comment]
Cool, all done.

File: examples/jsm/exporters/GLTFExporter.js (hunk 1 above)
Comment (Mugen87) [reply examples/jsm/exporters/GLTFExporter.js:106]: Yes!

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**